flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/FlinkOperatorConfiguration.java

```diff
@@ -220,8 +227,14 @@ private static GenericRetry getRetryConfig(Configuration conf) {
                                 .toMillis())
                 .setIntervalMultiplier(
                         conf.getDouble(
-                                KubernetesOperatorConfigOptions
-                                        .OPERATOR_RETRY_INTERVAL_MULTIPLIER));
+                                KubernetesOperatorConfigOptions.OPERATOR_RETRY_INTERVAL_MULTIPLIER))
+                .setMaxInterval(maxInterval > 0 ? maxInterval : Long.MAX_VALUE);
```

Comment:
The default value `Long.MAX_VALUE` means, retry as often as you like? Any value than that means wait at least that amount of time in milliseconds?

Reply:
It's a bit tricky because the JOSDK treats Long.MAX_VALUE as unset...

Reply:
I don't really see this MAX_VALUE behaviour that you mention, in any case I changed this to -1 which is the right way to set infinite

Reply:
See here for default `UNSET_VALUE`: https://github.com/operator-framework/java-operator-sdk/blob/a3fb3ad9ef84ad60e456ac2b3714b070b37e055b/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/retry/GenericRetry.java#L103 

which is set to `Long.MAX_VALUE`: https://github.com/operator-framework/java-operator-sdk/blob/a3fb3ad9ef84ad60e456ac2b3714b070b37e055b/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/retry/GradualRetry.java#L19

Reply:
If set to `Long.MAX_VALUE`, this default will be applied: https://github.com/operator-framework/java-operator-sdk/blob/a3fb3ad9ef84ad60e456ac2b3714b070b37e055b/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/retry/GradualRetry.java#L16

Reply:
Oh I see what you meant, this wouldn't affect us as we are not using the `@GradualRetry` annotation, that is specific to that. In any case I removed this to be less confusing